### PR TITLE
TGUI-next Nanite Interface Overhaul + Dropdown component

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

I'm finally emerging from the tgui dungeon to give you every single nanite interface completely reworked in full tgui-next glory, and a dropdown component.
On top of that, nanite extra settings have received a huge backend refactor to require infinitely less copypasta and boilerplate. That's what the -600 lines is.

Program Hub:
image
Programmer:
image
Cloud Controller:
image
Chamber Control Console:
image
Remote:
image

## Why It's Good For The Game

Old interfaces were horrendous

## Changelog
:cl:
add: All nanite interfaces have been completely reworked for tgui-next
/:cl:

### CODE REVIEW

File: code/modules/research/nanites/nanite_programs/suppression.dm [hunk 1]
```diff
@@ -131,38 +134,19 @@
 	trigger_cost = 3
 	trigger_cooldown = 20
 	rogue_types = list(/datum/nanite_program/brain_misfire, /datum/nanite_program/brain_decay)
+	var/static/list/blacklist = ("*surrender")
```
Comment (XDTM) [inline code/modules/research/nanites/nanite_programs/suppression.dm]: Make sure you document that you've enabled emotes in forced speech

Comment (actioninja) [reply code/modules/research/nanites/nanite_programs/suppression.dm]: fair enough, I just wanted to put in *clap

Comment (XDTM) [reply code/modules/research/nanites/nanite_programs/suppression.dm]: reminder to remove *collapse

**Comment by XDTM** [comment]
The programmer interface seems to be missing the trigger cooldown info

**Review by stylemistake — DISMISSED** [review]

**Review by stylemistake — CHANGES_REQUESTED** [review]
Fat fingered an approval, you need to rebase.

**Comment by actioninja** [comment]
DNM awaiting useBackend refactor for input fixes

**Comment by actioninja** [comment]
Ready.

**Review by stylemistake — DISMISSED** [review]

**Pull request merged**